<a href="https://colab.research.google.com/github/argONNY/-Untitled3.ipynb-/blob/main/lab1_notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv')

In [16]:
df.head(7)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [18]:
print(df.describe())
#Средний возраст - 30лет
#Макс цена за билет - 512.3292

       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  


In [19]:
#Часть 2

In [20]:
df['Age'] = df['Age'].fillna(df['Age'].median())

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [22]:
def get_age_group(age):
    if age < 18:
        return "Ребенок"
    elif age <= 65:
        return "Взрослый"
    else:
        return "Пожилой"
df['AgeGroup'] = df['Age'].apply(get_age_group)
print(df[['Age', 'AgeGroup']].head(10))

    Age  AgeGroup
0  22.0  Взрослый
1  38.0  Взрослый
2  26.0  Взрослый
3  35.0  Взрослый
4  35.0  Взрослый
5  28.0  Взрослый
6  54.0  Взрослый
7   2.0   Ребенок
8  27.0  Взрослый
9  14.0   Ребенок


In [23]:
#Часть 3

In [24]:
df.groupby('Sex')['Survived'].mean() * 100


Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64

In [25]:
df.groupby('Pclass')['Survived'].mean() * 100

Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64

In [26]:
survival_rate = df.groupby(['Sex', 'Pclass'])['Survived'].mean() * 100
print(survival_rate)

Sex     Pclass
female  1         96.808511
        2         92.105263
        3         50.000000
male    1         36.885246
        2         15.740741
        3         13.544669
Name: Survived, dtype: float64


In [27]:
result = (df[(df['AgeGroup'] == 'Ребенок') & (df['Pclass'] == 3) & (df['Survived'] == 1)].sort_values('Age', ascending=False)
['Name'])
print(result)

68                     Andersson, Miss Erna Alexandra
208                          Carr, Miss Helen "Ellen"
220                    Sunderland, Mr. Victor Francis
156                   Gilnagh, Miss Katherine "Katie"
22                         McGowan, Miss Anna "Annie"
875                   Najib, Miss Adele Kiamie "Jane"
830           Yasbeck, Mrs. Antoni (Selini Alexander)
39                         Nicola-Yarred, Miss Jamila
780                               Ayoub, Miss Banoura
125                       Nicola-Yarred, Master Elias
489              Coutts, Master Eden Leslie "Neville"
165    Goldsmith, Master Frank John William "Frankie"
751                                Moor, Master Meier
233                     Asplund, Miss Lillian Gertrud
448                     Baclini, Miss Marie Catherine
777                      Emanuel, Miss Virginia Ethel
10                     Sandstrom, Miss Marguerite Rut
869                    Johnson, Master Harold Theodor
691                         

In [28]:
oldest_male_not_survived = df[(df['Sex'] == 'male') & (df['Survived'] == 0)].sort_values(by='Age', ascending=False).iloc[0]
print(oldest_male_not_survived[['Name', 'Age']])

Name    Svensson, Mr. Johan
Age                    74.0
Name: 851, dtype: object


# Часть 4

1.Исследование демонстрирует, что гендер оказал наиболее заметное влияние на уровень выживаемости.
Доля выживших женщин составляет примерно 74%.
Среди представителей первого класса смогли спастись около 63%.

2.Полученные результаты подтверждают это утверждение.
Выживаемость женщин оказалась значительно выше, чем у мужчин (74% против 19%).
Дети младше 18 лет показывают уровень выживаемости около 50.85%.
(Выживаемость выше, чем у взрослых %36 и пожилых 12,5%)


